### PR TITLE
Housing edition

### PR DESCRIPTION
**Pull request by cveneziani**

### CODE REVIEW

File: apps/web/controllers/trips/housings/create.rb [hunk 1]
```diff
@@ -36,8 +36,8 @@ def load_trip
       end
 
       def authorize
-        authorization = Web::Trips::Housings::Authorization.new(current_user)
-        halt 401 unless authorization.create?(@trip)
+        authorization = Web::Trips::Housings::Authorization.new(current_user, @trip)
+        halt 401 unless authorization.create?
```
Comment (nfilzi) [inline apps/web/controllers/trips/housings/create.rb]: 👍

File: apps/web/controllers/trips/housings/update.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      end
+
+      private
+
+      def load_trip
+        @trip = TripRepository.new.find(params[:trip_id])
+      end
+
+      def load_housing
+        @housing = HousingRepository.new.find_for_trip(params[:id], params[:trip_id])
```
Comment (nfilzi) [inline apps/web/controllers/trips/housings/update.rb]: Do we need to use the `find_for_trip` to find the housing we are looking at with the `params[:id]` ? What's your reason for narrowing it down first by the `trip_id`?

Comment (cveneziani) [reply apps/web/controllers/trips/housings/update.rb]: I don't want the user to be able to provide a random id and then has access to a housing from someone else. What do you think?

Comment (nfilzi) [reply apps/web/controllers/trips/housings/update.rb]: Fine by me then!

Comment (cveneziani) [reply apps/web/controllers/trips/housings/update.rb]: Later on, I'll try to do only one query.

**Pull request merged**